hierarchicalforecast/core.py

```diff
@@ -5,18 +5,19 @@
 
 # %% ../nbs/src/core.ipynb 4
 import re
-import gc
 import time
 import copy
+
 from .methods import HReconciler
+from .utils import pivot
 from inspect import signature
+from narwhals.typing import FrameT
```

Comment:
I don't think `FrameT` is the right type here, TypeVars are useful when you want to indicate that the output type will be the same as one if the inputs, e.g. `def f(df: FrameT) -> FrameT`, that will tell type checkers that if you passed a pandas dataframe as `df`, you will get a pandas dataframe as output.

I believe what should be used here is either `Frame` or `IntoDataFrame`, depending on the case ([reference](https://narwhals-dev.github.io/narwhals/api-reference/typing/))

Reply:
Ah, very good point, fixed.